File: advance/problem2_aliceadd/aliceadd.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

string add(string a, string b) {
    if (a.length() < b.length()) { swap(a, b); }
    reverse(a.begin(), a.end());
    reverse(b.begin(), b.end());

    string result = "";
    int carry = 0;

    for (size_t i = 0; i < a.length(); ++i) {
        int da = a[i] - '0';
        int db = (i < b.length()) ? b[i] - '0' : 0;

        int sum = da + db + carry;
        result += (sum % 10) + '0';
        carry = sum/10;
    }

    if (carry > 0) {
        result += carry + '0';
    }

    reverse(result.begin(), result.end());
    return result;
} //sử dụng hàm add(...) ở Problem 1: ADD

int main() {
    int T;
    cin >> T;


    while (T--) {
        string a, b;
        cin >> a >> b;
        cout << add(a, b) << "\n";
    }
    return 0;
}

File: advance/problem2_aliceadd/aliceadd.go
package main

import (
	"fmt"
)

func reverse(s string) string {
	runes := []rune(s)
	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
		runes[i], runes[j] = runes[j], runes[i]
	}
	return string(runes)
}

func add(a, b string) string {
	a = reverse(a)
	b = reverse(b)

	if len(a) < len(b) {
		a, b = b, a
	}

	result := ""
	carry := 0

	for i := 0; i < len(a); i++ {
		digitA := int(a[i] - '0')
		digitB := 0
		if i < len(b) {
			digitB = int(b[i] - '0')
		}

		sum := digitA + digitB + carry
		carry = sum / 10
		result += string('0' + sum%10)
	}

	if carry > 0 {
		result += "1"
	}

	return reverse(result)
}

func main() {
	var T int
	fmt.Scan(&T)

	for i := 0; i < T; i++ {
		var a, b string
		fmt.Scan(&a, &b)
		fmt.Println(add(a, b))
	}
}
